gui/wxpython/history/tree.py

```diff
@@ -40,6 +41,46 @@
 from grass.grassdb import history
 
 
+# global variable for purposes of sorting "No time info" node
+OLD_DATE = datetime.datetime(1950, 1, 1).date()
+
+
+class HistoryBrowserNode(DictFilterNode):
+    """Node representing item in history browser."""
+
+    def __init__(self, data=None):
+        super().__init__(data=data)
+
+    @property
+    def label(self):
+        if "day" in self.data.keys():
+            return self.dayToLabel(self.data["day"])
+        else:
+            return self.data["name"]
+
+    def dayToLabel(self, day):
+        """
+        Convert day (midnight timestamp) to a day node label.
+        :param day datetime.date object: midnight of a day.
+        :return str: Corresponding day label.
+        """
+        current_date = datetime.date.today()
+
+        if day == OLD_DATE:
+            return _("No time info")
+
+        if day == current_date:
+            return "{:%B %-d} (today)".format(day)
+        elif day == current_date - datetime.timedelta(days=1):
+            return "{:%B %-d} (yesterday)".format(day)
+        elif day >= current_date - datetime.timedelta(days=7):
+            return "{:%B %-d} (this week)".format(day)
```

Comment:
I think this week should refer to the Mon-Sun week, not past 7 days. Looks like iso standard takes Monday as the first day of the week (not Sunday as in the US).

Reply:
One other suggestion is to not show the year for current year to improve readability, but feel free to ignore this.

Reply:
> One other suggestion is to not show the year for current year to improve readability, but feel free to ignore this.

Yes, it is better not to show the current year. Btw I also removed the (this month) label as I suggested above - in this case the browser simply shows just the format 'December 21'.
![Screenshot from 2024-05-02 09-19-57](https://github.com/OSGeo/grass/assets/49241681/96feb5e9-d98f-4feb-89c8-88547c53af41)

Reply:
I think this PR is ready for the final review :-).